ci/Dockerfile

```diff
@@ -64,15 +64,14 @@ RUN set -e; \
             GOBIN_VERSION='arm64'; \
             ;; \
         *) echo >&2 "error: unsupported architecture: '$ARCH_NAME'"; exit 1 ;; \
-    esac; \
-    microdnf module enable php ruby -y \
+    esac \
     && microdnf install -y php php-curl php-zip php-bcmath php-json php-pear php-mbstring php-devel make gcc git-core \
         python3.11 python3.11-devel python3.11-pip ruby ruby-devel glibc-common glibc-all-langpacks \
         pcre2 which tar gzip zip unzip sudo ncurses sqlite-devel dotnet-sdk-8.0 \
     && alternatives --install /usr/bin/python3 python /usr/bin/python3.11 1 \
-    && python3 --version \
-    && python3 -m pip install --upgrade pip virtualenv \
-    && python3 -m pip install --user pipenv poetry blint \
+    && python3.11 --version \
```

Comment:
How about we offer `PYTHON_VERSION` as a [build arg](https://docs.docker.com/build/guide/build-args/) and refer it here and in the install step above. Let's use 3.12 and bump the version to 10.6.x